From 6eed9cd36808e77a18acf554bb5753978906c1aa Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Fri, 11 Mar 2011 21:01:24 +0100
Subject: [PATCH] Add paths to qt plugins (from kstandarddirs) before qapp is created.

Necessary for apps started through d-bus activation, so that the kde widget
style is loaded.
FIXED-IN: 4.6.2
BUG: 267770
---
 kdecore/kernel/kcomponentdata.cpp |    2 +-
 kdeui/kernel/kapplication.cpp     |    2 ++
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/kdecore/kernel/kcomponentdata.cpp b/kdecore/kernel/kcomponentdata.cpp
index 38bda18..933623a 100644
--- a/kdecore/kernel/kcomponentdata.cpp
+++ b/kdecore/kernel/kcomponentdata.cpp
@@ -144,7 +144,7 @@ void KComponentDataPrivate::lazyInit(const KComponentData &component)
     }
     
     // the first KComponentData sets the KDE Qt plugin paths
-    if (QCoreApplication::instance() && dirs && kdeLibraryPathsAdded != KdeLibraryPathsAddedDone) {
+    if (dirs && kdeLibraryPathsAdded != KdeLibraryPathsAddedDone) {
         kdeLibraryPathsAdded = KdeLibraryPathsAddedDone;
         const QStringList &plugins = dirs->resourceDirs("qtplugins");
         QStringList::ConstIterator it = plugins.begin();
diff --git a/kdeui/kernel/kapplication.cpp b/kdeui/kernel/kapplication.cpp
index 1403601..8ec47ca 100644
--- a/kdeui/kernel/kapplication.cpp
+++ b/kdeui/kernel/kapplication.cpp
@@ -407,6 +407,8 @@ KApplication::KApplication(Display *display, int& argc, char** argv, const QByte
 void KApplicationPrivate::preqapplicationhack()
 {
     preread_app_startup_id();
+
+    KGlobal::config(); // initialize qt plugin path (see KComponentDataPrivate::lazyInit)
 }
 
 int KApplication::xioErrhandler( Display* dpy )
-- 
1.7.1

